sdk/metric/meter.go

```diff
@@ -60,9 +66,15 @@ var _ metric.Meter = (*meter)(nil)
 // options. The instrument is used to synchronously record increasing int64
 // measurements during a computational operation.
 func (m *meter) Int64Counter(name string, options ...metric.Int64CounterOption) (metric.Int64Counter, error) {
+
 	cfg := metric.NewInt64CounterConfig(options...)
 	const kind = InstrumentKindCounter
-	return m.int64IP.lookup(kind, name, cfg.Description(), cfg.Unit())
+	i, err := m.int64IP.lookup(kind, name, cfg.Description(), cfg.Unit())
+	if err != nil {
+		return i, err
+	}
+
+	return i, validateInstrumentName(name)
```

Comment:
Moving the name validation to the instrument provider lookup (e.g. https://github.com/open-telemetry/opentelemetry-go/blob/5d57493de9448db306da915c6c900e9aaf2ce172/sdk/metric/meter.go#L436-L440) will centralize this logic into two places instead of all the instrument methods.

Reply:
See ec8ccc5fa00b84f24b63e48714672a8ee8e3e5d2, which I'm not convinced is simpler.

Reply:
> See [ec8ccc5](https://github.com/open-telemetry/opentelemetry-go/commit/ec8ccc5fa00b84f24b63e48714672a8ee8e3e5d2), which I'm not convinced is simpler.

I see your point. I didn't consider the need to validate the returned error as being `ErrInvalidInstrumentName`.

I'm fine with reverting if you think the original approach better. Sorry for the poor suggestion.